src/commons/eventsourcing/wal.rs

```diff
@@ -209,8 +225,17 @@ impl<T: WalSupport> WalStore<T> {
         if !self.has(handle)? {
             Err(WalStoreError::Unknown(handle.clone()))
         } else {
-            self.cache.write().unwrap().remove(handle);
-            self.kv.drop_scope(handle.as_str())?;
+            {
+                // First get a lock and remove the object
+                let handle_lock = self.locks.for_handle(handle.clone());
+                let _write = handle_lock.write();
+                self.cache.write().unwrap().remove(handle);
+                self.kv.drop_scope(handle.as_str())?;
+            }
+
+            // Then drop the lock for it as well. We could not do this
+            // while holding the write lock. Note that
```

Comment:
Incomplete sentence.

Reply:
Ah yes, oops.. I remember now that as I was typing a comment that was going to say that a race condition would be unlikely I figured that perhaps.. we should have a `self.locks.remove_handle` instead. Let me take a stab at that in a moment..

Reply:
updated the comment